pytorch_lightning/trainer/trainer.py

```diff
@@ -1520,8 +1520,7 @@ def _call_lightning_module_hook(
         prev_fx_name = pl_module._current_fx_name
         pl_module._current_fx_name = hook_name
 
-        # TODO: when profiling separate hook name by hook object name (e.g. Callback, LM)
-        with self.profiler.profile(hook_name):
+        with self.profiler.profile(f"{pl_module.__class__.__name__}.{hook_name}"):
```

Comment:
Do we need the prefix to be deterministic? Right now if we have `MyCustomLightningModule.{hook_name}` it can be hard for the profiler to distinguish this hook vs `MyCustomPlugin.{hook_name}` 

Something like `LightningModule.MyCustomLightningModule.{hook_name}` could provide a descriptive record and allow downstream systems to better process this data, at the cost of verbosity

Reply:
Makes sense! will update